k8s-deployment-manifest-templates/deployment-mode/daemonset/container-insights-monitoring/fluent-bit/fluent-bit.yaml

```diff
@@ -71,7 +71,7 @@ data:
         Docker_Mode         On
         Docker_Mode_Flush   5
         Docker_Mode_Parser  container_firstline
-        Parser              docker
+        Parser              docker, cri
```

Comment:
I think this will not work. 

See here: https://docs.fluentbit.io/manual/pipeline/inputs/tail#multiline-and-containers-v1.8

We use the multiline parser. And then you need to remove the old Docker_mode options: https://docs.fluentbit.io/manual/pipeline/inputs/tail#docker_mode

If you notice here in the Fluent Bit upstream official helm chart, they just use the multiline.parser, nothing more: https://github.com/fluent/helm-charts/blob/main/charts/fluent-bit/values.yaml#L313

@SaxyPandaBear I think we possibly need to deploy this to a real cluster just to double check that the parsing is not broken. Are you able to do that?

Reply:
🤦 Totally messed that up. Thanks for the catch. 

Yeah I can make the fix and deploy to smoke test

Reply:
So I tested a simple input config like you see in my link in ECS with Docker. This obviously does not directly apply here since k8s is now using cri:

```
[INPUT]
    Name                tail
    Tag                 app.*
    Path                /var/lib/docker/containers/*/*.log
    multiline.parser    docker, cri
```

This parses my logs happily, no need for a `Parser docker` entry. 

I used this image test which you can use as well @SaxyPandaBear. It outputs logs once per second or so that are greater than 16KB such that they should get split by the runtime stdout pipe, so it tests that FLB re-joins them together properly:

```
144718711470.dkr.ecr.us-west-2.amazonaws.com/multiline-long-lines:latest
```

Reply:
(I also double checked that timestamp parsing from the log file works correctly with the above config). (Again, this is with Docker, not CRI).

Reply:
Finally figured out what was wrong with my test cluster. I ran what we have currently, and it seems to break up the log events with stdout piping?

Grouped log events start with:
```
{"log":"2023-01-09T16:07:53.012056409Z stdout P {\"payload\": \"01234567890abcdefghijklmnopqrstuvwxyz01234567890

{"log": "2023-01-09T16:07:53.013404323Z stdout P ghijklmnopqrstuvwxyz01234567890abcdefghij

// etc
```

and it's broken up into 5 events. I'm assuming that's expected since that's with the current config from the quickstart yaml. 

Going to figure out how to force CRI and then redo the test with my changes.

Reply:
Ah okay I see. So maybe that was broken before..? Or I messed up my config. With my change, it's definitely combining into a single log event. Will push a new revision

Reply:
I think what you have currently might not do multiline for CRI, only for Docker. So since these logs are from CRI, only with the new CRI multiline parser are they getting recombined. 

Long log messages are split by container runtimes at 16KB, and then the multiline parser can rejoin them.

Your example is simplified right? Its showing that the log is split right?

Reply:
Right. I didn't paste the entire log content, just the beginning

Reply:
After I deployed with the multiline + CRI parser change, I could see very clearly that the logs I saw earlier were not what we want, and making the CRI change fixes it. There's no partial log entry after the change

Reply:
Cool!